### PR TITLE
Don't dump textures that aren't a power of 2

### PR DESCRIPTION
**Pull request by khang06**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5152)

### CODE REVIEW

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 1]
```diff
@@ -884,45 +884,51 @@ bool CachedSurface::LoadCustomTexture(u64 tex_hash, Core::CustomTexInfo& tex_inf
...
-        fmt::format("{}textures/{:016X}/", FileUtil::GetUserPath(FileUtil::UserPath::DumpDir),
-                    Core::System::GetInstance().Kernel().GetCurrentProcess()->codeset->program_id);
-    if (!FileUtil::CreateFullPath(dump_path)) {
-        LOG_ERROR(Render, "Unable to create {}", dump_path);
-        return;
-    }
+    // Make sure the texture size is a power of 2
+    std::bitset<32> width_bits(width);
+    std::bitset<32> height_bits(height);
+    if (width_bits.count() == 1 && height_bits.count() == 1) {
```
Comment (vitor-k) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: This condition can be inverted to unindent the code by one level with an early return.

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 2]
```diff
@@ -884,6 +884,15 @@ bool CachedSurface::LoadCustomTexture(u64 tex_hash, Core::CustomTexInfo& tex_inf
 }
 
 void CachedSurface::DumpTexture(GLuint target_tex, u64 tex_hash) {
+    // Make sure the texture size is a power of 2
+    std::bitset<32> width_bits(width);
+    std::bitset<32> height_bits(height);
```
Comment (zhaowenlan1779) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: `<bitset>` should be included.

File: src/video_core/renderer_opengl/gl_rasterizer_cache.cpp [hunk 3]
```diff
@@ -905,17 +914,16 @@ void CachedSurface::DumpTexture(GLuint target_tex, u64 tex_hash) {
...
+            GetTexImageOES is used even if not using OpenGL ES to work around a small issue that
+            happens if using custom textures with texture dumping at the same.
+            Let's say there's 2 textures that are both 32x32 and one of them gets replaced with a
+            higher quality 256x256 texture. If the 256x256 texture is displayed first and the
+            32x32 texture gets uploaded to the same underlying OpenGL texture, the 32x32 texture
+            will appear in the corner of the 256x256 texture. If texture dumping is enabled and
+            the 32x32 is undumped, Citra will attempt to dump it. Since the underlying OpenGL
+            texture is still 256x256, Citra crashes because it thinks the texture is only 32x32.
+            GetTexImageOES conveniently only dumps the specified region, and works on both
+            desktop and ES.
```
Comment (zhaowenlan1779) [inline src/video_core/renderer_opengl/gl_rasterizer_cache.cpp]: This comment block should be left unchanged

**Comment by lioncash** [comment]
First commit should probably explain why they shouldn't be dumped, just for clarification if the commit is ever part of a blame lookup in the future.

**Pull request merged**